store/mockstore/mockstore.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mockstore
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pingcap/errors"
+	pd "github.com/pingcap/pd/v4/client"
+	"github.com/pingcap/tidb/config"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/store/mockstore/cluster"
+	"github.com/pingcap/tidb/store/mockstore/mocktikv"
+	"github.com/pingcap/tidb/store/mockstore/unistore"
+	"github.com/pingcap/tidb/store/tikv"
+)
+
+// MockTiKVDriver is in memory mock TiKV driver.
+type MockTiKVDriver struct{}
+
+// Open creates a MockTiKV storage.
+func (d MockTiKVDriver) Open(path string) (kv.Storage, error) {
+	u, err := url.Parse(path)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !strings.EqualFold(u.Scheme, "mocktikv") {
+		return nil, errors.Errorf("Uri scheme expected(mocktikv) but found (%s)", u.Scheme)
+	}
+
+	opts := []MockTiKVStoreOption{WithPath(u.Path), WithStoreType(MockTiKV)}
+	txnLocalLatches := config.GetGlobalConfig().TxnLocalLatches
+	if txnLocalLatches.Enabled {
+		opts = append(opts, WithTxnLocalLatches(txnLocalLatches.Capacity))
+	}
+
+	return NewMockStore(opts...)
+}
+
+// EmbedUnistoreDriver is in embedded unistore driver.
+type EmbedUnistoreDriver struct{}
+
+// Open creates a EmbedUnistore storage.
+func (d EmbedUnistoreDriver) Open(path string) (kv.Storage, error) {
+	u, err := url.Parse(path)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !strings.EqualFold(u.Scheme, "unistore") {
+		return nil, errors.Errorf("Uri scheme expected(unistore) but found (%s)", u.Scheme)
+	}
+
+	opts := []MockTiKVStoreOption{WithPath(u.Path), WithStoreType(EmbedUnistore)}
+	txnLocalLatches := config.GetGlobalConfig().TxnLocalLatches
+	if txnLocalLatches.Enabled {
+		opts = append(opts, WithTxnLocalLatches(txnLocalLatches.Capacity))
+	}
+
+	return NewMockStore(opts...)
+}
+
+// StoreType is the type of backend mock storage.
+type StoreType uint8
+
+const (
+	// MockTiKV is the mock storage based on goleveldb.
+	MockTiKV StoreType = iota
+	// EmbedUnistore is the mock storage based on unistore.
+	EmbedUnistore
+
+	defaultStoreType = EmbedUnistore
```

Comment:
I think it is better to use `MockTiKV` as the default storage backend, and add a new CI task to use unistore (how to achieve this? use env variables?).
If the new CI task is stable enough, we can remove the mocktikv when the next time we need to implement features on mocktikv.